cmd/common/frontends/console/console.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package console
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/term"
+
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/common/frontends"
+	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+)
+
+type frontend struct {
+	ctx    context.Context
+	cancel func()
+}
+
+func NewFrontend() frontends.Frontend {
+	ctx, cancel := context.WithCancel(context.Background())
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-stop
+		log.Debugf("got signal, cancelling context")
+		cancel()
+		<-stop
+		log.Debugf("got signal again, exiting")
+		// On second signal, exit forcefully
+		os.Exit(1)
+	}()
+	return &frontend{
+		ctx:    ctx,
+		cancel: cancel,
+	}
+}
+
+func (f *frontend) Clear() {
+	if term.IsTerminal(int(os.Stdout.Fd())) {
+		commonutils.ClearScreen()
+	}
+}
+
+func (f *frontend) Output(payload string) {
+	fmt.Fprintln(os.Stdout, payload)
+}
+
+func (f *frontend) Error(severity logger.Level, message string) {
+	fmt.Fprintln(os.Stderr, message)
+}
```

Comment:
I'm wondering if this should be called `Log()` instead and us the a logger underneath? (I can work on it if you think it makes sense).

Reply:
Yeah, makes sense - feel free to change it!

Reply:
Implemented in https://github.com/inspektor-gadget/inspektor-gadget/commit/4cd4eb38230e389fdb8022681200664282d89e57 and https://github.com/inspektor-gadget/inspektor-gadget/commit/32d485b07a9010b93841e676a8968c7d39c2b51f.